### PR TITLE
HEEDLS-685 show professional registration number on My Account page

### PR DESCRIPTION
**Pull request by rhimcg**
### JIRA link
_HEEDLS-685_(https://softwiretech.atlassian.net/jira/software/c/projects/HEEDLS/boards/671?modal=detail&selectedIssue=HEEDLS-685&assignee=6065c334b30f0d00704264d4)

### Description
Add the professional registration number to the my account page. 

- Shows "Not yet registered" if they have not yet been asked for their PRN. 
- "Not professionally registered" if they have been asked for their PRN but it is null
- And the professional registration number if they have been asked for and provided one

### Screenshots
hee685 prn
hee685 not registered
hee685 not yet

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [ ] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [ ] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs [hunk 1]
```diff
@@ -23,12 +25,24 @@ public class MyAccountViewModel
             DelegateNumber = delegateUser?.CandidateNumber;
             AliasId = delegateUser?.AliasId;
             JobGroup = delegateUser?.JobGroupName;
+            ProfessionalRegistrationNumber = delegateUser?.HasBeenPromptedForPrn == true
+                ? delegateUser.ProfessionalRegistrationNumber ?? "Not professionally registered"
+                : "Not yet provided";
+
+            Console.WriteLine(delegateUser);
```
Comment (davidm-m) [inline DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs:32]: I'm guessing this wasn't meant to be left in 😄

File: DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.MyAccount
 {
+    using System;
```
Comment (davidm-m) [inline DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs:3]: Once you've got rid of the `WriteLine` you won't need this import either.

**Review by davidm-m — CHANGES_REQUESTED** [review]
You've accidentally left in a `Console.WriteLine`, otherwise looks great.

**Review by davidm-m — APPROVED** [review]
Great stuff

File: DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs [hunk 3]
```diff
@@ -23,12 +24,22 @@ public class MyAccountViewModel
             DelegateNumber = delegateUser?.CandidateNumber;
             AliasId = delegateUser?.AliasId;
             JobGroup = delegateUser?.JobGroupName;
+            ProfessionalRegistrationNumber = delegateUser?.HasBeenPromptedForPrn == true
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs:27]: Can you get rid of the `== true`? You might need a `.Value`. That's the more conventional way of doing it when you have a bool variable.

File: DigitalLearningSolutions.Web.Tests/ViewModels/MyAccount/MyAccountViewModelTests.cs [hunk 4]
```diff
@@ -123,5 +131,83 @@ public void MyAccountViewModel_CustomFields_ShouldBePopulated()
...
             }
         }
+
+        [Test]
+        public void MyAccountViewModel_where_user_has_not_been_asked_for_prn_says_not_yet_provided()
+        {
+            // Given
+            var delegateUser = UserTestHelper.GetDefaultDelegateUser(
+                hasBeenPromptedForPrn: false,
+                professionalRegistrationNumber: "123456"
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Web.Tests/ViewModels/MyAccount/MyAccountViewModelTests.cs:141]: I think it makes more sense to test the case where `professionalRegistrationNumber` is null, because that's what we expect to see in production, and it's entirely possible that `professionalRegistrationNumber: null` would override `hasBeenPromptedForPrn: false`.

File: DigitalLearningSolutions.Web.Tests/ViewModels/MyAccount/MyAccountViewModelTests.cs [hunk 5]
```diff
@@ -123,5 +131,83 @@ public void MyAccountViewModel_CustomFields_ShouldBePopulated()
...
+            );
+            var customPrompts = CustomPromptsTestHelper.GetDefaultCentreCustomPromptsWithAnswers(
+                new List<CustomPromptWithAnswer>
+                {
+                    CustomPromptsTestHelper.GetDefaultCustomPromptWithAnswer(1),
+                    CustomPromptsTestHelper.GetDefaultCustomPromptWithAnswer(2),
+                }
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Web.Tests/ViewModels/MyAccount/MyAccountViewModelTests.cs:148]: Can we not just make `customPrompts` an empty list?

File: DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs (hunk 3 above)
Comment (rhimcg) [reply DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs:27]: its telling me .Value doesn't exist on HasBeenPrompted and not letting me do .Equals because it could be null as delegateUser is potentially null

Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs:27]: Oh, I see, HasBeenPromptedForPrn isn't nullable, it's because delegateUser is nullable. Okay, that's fine.

**Review by SteveJacksonSoft — APPROVED** [review]
Merge when ready.
